p2p/discover/v4_udp.go

```diff
@@ -387,23 +410,34 @@ func (t *UDPv4) lookupWorker(n *node, targetKey encPubkey, reply chan<- []*node)
 	reply <- r
 }
 
-// Resolve searches for a specific node with the given ID.
-// It returns nil if the node could not be found.
+// Resolve searches for a specific node with the given ID and tries to get the most recent
+// version of the node record for it. It returns n if the node could not be resolved.
 func (t *UDPv4) Resolve(n *enode.Node) *enode.Node {
-	// If the node is present in the local table, no
-	// network interaction is required.
-	if intab := t.tab.Resolve(n); intab != nil {
-		return intab
-	}
-	// Otherwise, do a network lookup.
-	hash := n.ID()
-	result := t.LookupPubkey(n.Pubkey())
-	for _, n := range result {
-		if n.ID() == hash {
-			return n
+	// Try asking directly. This works if the node is still responding on the endpoint we have.
+	if rn, err := t.requestENR(n); err == nil {
+		return rn
+	}
```

Comment:
This code modifies the behavior slightly, whereby previously we always used the local database if we've seen the node before, but now we always reach out to the network to do an ENR request. Are we sure there's no issue in this new network request based lookup?

Reply:
There is no issue with the request because it will deliver the most up-to-date version of the record if it works. If the request fails we just do what we did before.

Reply:
I was more concerned about the introduced network RTT.